Observer/Order/Save.php

```diff
@@ -148,6 +152,16 @@ public function execute(Observer $observer)
                 if ($nostoCustomer instanceof NostoCustomer) {
                     $nostoCustomerId = $nostoCustomer->getNostoId();
                 }
+                // If the id is still null, fetch the `customer_reference`
+                if ($nostoCustomerId === null) {
+                    $customerId = $order->getCustomerId();
+                    $magentoCustomer = $this->magentoCustomerRepository->getById($customerId);
+                    // Get the value of `customer_reference`
+                    $customerReferenceAttribute = $magentoCustomer->getCustomAttribute(
+                        NostoHelperData::NOSTO_CUSTOMER_REFERENCE_ATTRIBUTE_NAME
+                    );
+                    $nostoCustomerId = $customerReferenceAttribute->getValue();
```

Comment:
```
$magentoCustomer->getCustomAttribute(
    NostoHelperData::NOSTO_CUSTOMER_REFERENCE_ATTRIBUTE_NAME
);
```
This can return null, which can cause you to call a function from null.